dbcon/mysql/ha_mcs_client_udfs.cpp

```diff
@@ -924,6 +924,103 @@ extern "C"
   {
   }
 
+  // --- Plan UDFs ---
+  // Return plan strings or applied rules based on optional parameter:
+  //   no arg or 1 -> optimized/latest plan (ci->queryPlan)
+  //   0 or 'original'  -> pre-RBO plan
+  //   1 or 'optimized' -> post-RBO plan
+  //   2 or 'rules'     -> comma-separated applied RBO rules
+  const char* mcsgetplan(UDF_INIT* /*initid*/, UDF_ARGS* args, char* /*result*/, unsigned long* length,
+                         char* is_null, char* /*error*/)
+  {
+    if (get_fe_conn_info_ptr() == NULL)
+    {
+      set_fe_conn_info_ptr((void*)new cal_connection_info());
+      thd_set_ha_data(current_thd, mcs_hton, get_fe_conn_info_ptr());
+    }
+
+    cal_connection_info* ci = reinterpret_cast<cal_connection_info*>(get_fe_conn_info_ptr());
+    static const std::string wrongArg("Wrong argument produced");
+    const std::string* out = &wrongArg;
+    ;
+    if (args && args->arg_count >= 1 && args->args[0])
+    {
+      if (args->arg_type[0] == INT_RESULT)
```

Comment:
I am missing what is the point of using numbers instead of strings only here?

Reply:
Shorter to type in the interactive mode